libraries/AP_HAL_SITL/Scheduler.cpp

```diff
@@ -130,7 +130,9 @@ void Scheduler::sitl_end_atomic() {
 
 void Scheduler::reboot(bool hold_in_bootloader)
 {
-    hal.uartA->printf("REBOOT NOT IMPLEMENTED\r\n\n");
+    execv(_sitlState->_argv[0], _sitlState->_argv);
```

Comment:
lucasdemarchi 12 minutes ago  Member
It would be nice to have the exec in the same place as the mainloop, finalizing all components before. Although for sitl this may indeed be sufficient

  @khancyr
khancyr 5 minutes ago  Member
you mean at HAL_SITL::run level and catching an exit() comming from the Scheduler::reboot() function?

Reply:
lucasdemarchi : 
Having a flag _should_exit and looping on it. I vaguely remember doing that on Linux Hal because we need to do some teardown() calls (e.g. to shutdown the dma handling on RPI)